frame/staking-rewards/src/lib.rs

```diff
@@ -228,20 +252,21 @@ pub mod pallet {
 	>;
 
 	#[pallet::type_value]
-	pub fn SharesOnEmpty<T: Config>() -> u128 {
-		u128::zero()
+	pub fn SharesOnEmpty<T: Config>() -> ShareAmount {
+		ShareAmount::zero()
 	}
 
+	//// active running total shares
```

Comment:
Would be nice to have better docs on why this storage exists

Reply:
total stake is depend on this. you should not modify total stake until reward happend.

will add docs in next PRs